src/Workleap.OpenApi.MSBuild/SwaggerManager.cs

```diff
@@ -0,0 +1,69 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SwaggerManager : ISwaggerManager
+{
+    private const string SwaggerVersion = "6.5.0";
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiWebApiAssemblyPath;
+    private readonly string _swaggerDirectory;
+
+    public SwaggerManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, string openApiWebApiAssemblyPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiWebApiAssemblyPath = openApiWebApiAssemblyPath;
+        this._swaggerDirectory = Path.Combine(openApiToolsDirectoryPath, $"swagger/{SwaggerVersion}");
+    }
+
+    public async Task RunSwaggerAsync(string[] openApiSwaggerDocumentNames, CancellationToken cancellationToken)
+    {
+        var swaggerExePath = Path.Combine(this._swaggerDirectory, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "swagger.exe" : "swagger");
+
+        foreach (var documentName in openApiSwaggerDocumentNames)
+        {
+            var outputOpenApiSpecName = $"openapi-{documentName.ToLowerInvariant()}.yaml";
+
+            var outputOpenApiSpecPath = Path.Combine(this._swaggerDirectory, outputOpenApiSpecName);
+
+            await this.GenerateOpenApiSpecAsync(swaggerExePath, outputOpenApiSpecPath, documentName, cancellationToken);
+        }
+    }
+
+    public async Task InstallSwaggerCliAsync(CancellationToken cancellationToken)
```

Comment:
Could we check if tool already exists instead of do a download for nothing?
Also this involves that if the download fails => we must cleanup the eventual partial created file to prevent the "does the tool already exists" check to return a false positive

Reply:
We do an dotnet update so no download is done for swagger if already installed.